### PR TITLE
chore: replaces `tx.get::<Table>` with provider methods

### PR DESCRIPTION
**Pull request by joshieDo**
As the title suggests. Replaces some low hanging fruit. There's some other uses, but it requires changing a few other things that I'd rather save for another PR. (eg. `assert_db` on ef-tests)

~~Waiting on https://github.com/paradigmxyz/reth/pull/3168  to be merged, so I can replace a `.get::<TxSenders` for its newly created provider function.~~

### CODE REVIEW

File: bin/reth/src/node/mod.rs [hunk 1]
```diff
@@ -509,30 +505,32 @@ impl Command {
...
-                difficulty: header.difficulty,
-                total_difficulty: total_difficulty.into(),
-                timestamp: header.timestamp,
-            })
-        })?
-        .map_err(Into::into)
+    fn lookup_head(&self, db: Arc<Env<WriteMap>>) -> Result<Head, reth_interfaces::Error> {
+        let factory = ProviderFactory::new(db, self.chain.clone());
+        let provider = factory.provider()?;
+        let head = get_stage_checkpoint(provider.tx_ref(), StageId::Finish)?
```
Comment (gakonst) [inline bin/reth/src/node/mod.rs:511]: cant we do provider.tx-ref().get_stage_checkpoint?

**Review by gakonst — COMMENTED** [review]
starting to look really clean - wdyt @onbjerg

**Review by gakonst — COMMENTED** [review]
Is your intent to test these provider methods as you port them? Or do you intend to add tests to each provider method afterwards?

**Comment by joshieDo** [comment]
this one PR is just replacing the current usage `.get::<table> -> provider.method` 

for actually porting the categorized ones, i'll try to come up with meaningful tests

**Review by gakonst — APPROVED** [review]

**Comment by gakonst** [comment]
roger - nice work!

**Pull request merged**